### PR TITLE
refactor: do not pass private key to Sign() of VC / VP

### LINKED ISSUE #339: Refactoring creation of serialized JWT from Verifiable Credential
In verifiable.Credential.JWT() method, the private key is passed as parameter. According to @llorllale, 
>Private keys will never leave the wallet, therefore they cannot be passed in as a parameter to anything outside of the wallet itself. 
Therefore, refactoring is needed here.

### PR DESCRIPTION
**Pull request by kdimak**
closes #339

Signed-off-by: Dmitriy Kinoshenko <dkinoshenko@gmail.com>

We start to use own impl of JWT / JWS in `verifiable`.
Internal impl of unsecured JWT is removed from `verifiable`. It was present due to no support in go-jose.

### CODE REVIEW

**Comment by troyronda** [comment]
@kdimak please rebase.

File: test/bdd/pkg/verifiable/verifiable_steps.go [hunk 1]
```diff
@@ -147,16 +136,21 @@ func (s *SDKSteps) addVCProof(vc *verifiable.Credential, issuer, proofType strin
 }
 
 func (s *SDKSteps) verifyCredential(holder string) error {
```
Comment (kdimak) [inline test/bdd/pkg/verifiable/verifiable_steps.go]: Here we use the same approach for VC verification. `WithPublicKeyFetcher()` is required for both JWS and Linked Data Proof verification. `WithEmbeddedSignatureSuites()` is LDP-specific.

File: test/bdd/pkg/verifiable/verifiable_steps.go [hunk 2]
```diff
@@ -103,38 +100,30 @@ func (s *SDKSteps) addVCProof(vc *verifiable.Credential, issuer, proofType strin
...
-		pubKey, privKey, err := ed25519.GenerateKey(rand.Reader)
-		if err != nil {
-			return nil, err
-		}
-
-		// keep pubKey to be used later for verification by a Holder
-		s.pubKey = pubKey
-
-		jws, err := jwtClaims.MarshalJWS(verifiable.EdDSA, privKey, "key-1")
+		jws, err := jwtClaims.MarshalJWS(verifiable.EdDSA, signer, pubKey.ID)
```
Comment (kdimak) [inline test/bdd/pkg/verifiable/verifiable_steps.go]: Here we just pass `signer` (not a private key). Note that the same `signer` is used for LDP and JWS.

File: pkg/doc/jwt/jwt.go [hunk 3]
```diff
@@ -209,24 +180,90 @@ func mapJWSToJWT(jws *jose.JSONWebSignature) (*JSONWebToken, error) {
...
+		isValidJSON(parts[0]) &&
+		isValidJSON(parts[1]) &&
+		parts[2] != ""
+}
+
+// IsJWTUnsecured checks if JWT is an unsecured JWT of valid structure.
+func IsJWTUnsecured(s string) bool {
+	parts := strings.Split(s, ".")
+
+	isValidJSON := func(s string) bool {
```
Comment (Baha-sk) [inline pkg/doc/jwt/jwt.go]: this variable can be moved out and used by both `IsJWTUnsercured()` and `IsJWS()`

File: pkg/doc/verifiable/credential_jwt_test.go [hunk 4]
```diff
@@ -10,12 +10,15 @@ import (
 	"testing"
 	"time"
 
-	"github.com/square/go-jose/v3/jwt"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/jwt"
```
Comment (Baha-sk) [inline pkg/doc/verifiable/credential_jwt_test.go]: double check import order .. usually Aries-framework-go should be the last import line

File: pkg/doc/verifiable/credential_jwt_unsecured_test.go [hunk 5]
```diff
@@ -9,6 +9,8 @@ import (
 	"encoding/json"
 	"testing"
 
+	"github.com/hyperledger/aries-framework-go/pkg/doc/jose"
```
Comment (Baha-sk) [inline pkg/doc/verifiable/credential_jwt_unsecured_test.go]: move this line below third party imports

File: pkg/doc/verifiable/credential_test.go [hunk 6]
```diff
@@ -14,6 +14,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/hyperledger/aries-framework-go/pkg/doc/jose"
+
```
Comment (Baha-sk) [inline pkg/doc/verifiable/credential_test.go]: extra space... and move Aries-framework-go imports to the end

File: pkg/doc/verifiable/jwt_unsecured_test.go [hunk 7]
```diff
@@ -6,159 +6,37 @@ SPDX-License-Identifier: Apache-2.0
...
 
 import (
-	"encoding/base64"
-	"encoding/json"
-	"fmt"
 	"testing"
 
-	"github.com/stretchr/testify/require"
-)
+	"github.com/hyperledger/aries-framework-go/pkg/doc/jose"
```
Comment (Baha-sk) [inline pkg/doc/verifiable/jwt_unsecured_test.go]: move this import to the end after 3rd party import below

File: pkg/doc/verifiable/presentation_jwt_unsecured_test.go [hunk 8]
```diff
@@ -8,6 +8,8 @@ package verifiable
 import (
 	"testing"
 
+	"github.com/hyperledger/aries-framework-go/pkg/doc/jose"
+
```
Comment (Baha-sk) [inline pkg/doc/verifiable/presentation_jwt_unsecured_test.go]: idem for imports

File: pkg/doc/jwt/jwt.go (hunk 3 above)
Comment (kdimak) [reply pkg/doc/jwt/jwt.go]: Ah, right! I did this once but probably rewrote this change. Fixed.

**Comment by kdimak** [comment]
Updated the PR - applied code review comments by @Baha-sk . Thank you, @Baha-sk !

**Pull request merged**